### PR TITLE
Updating credentials to use a struct with more than location

### LINKED ISSUE #27: Spec Change: Credentials now have a required attribute
The CNAB spec was updated to introduce a [required](https://github.com/deislabs/cnab-spec/pull/189) attribute on credentials. More generally, we don't properly model the credential as it also has a description.  We should update cnab-go to reflect that, and introduce a Credential struct that models this correctly, instead of just using the Location type.

### PR DESCRIPTION
**Pull request by jeremyrickard**
Per the spec, credentials have more attributes than just location, so this PR changes the definition around. Originally, this was just adding the `required` field but we also didn't have description. 

Closes #27

### CODE REVIEW

File: bundle/credentials.go [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package bundle
+
+// Credential represents the definition of a CNAB credential
+type Credential struct {
+	Location    `json:",squash" mapstructure:",squash"`
```
Comment (silvin-lubecki) [inline bundle/credentials.go]: nit: I guess this file needs a `go fmt` 😸

Comment (silvin-lubecki) [inline bundle/credentials.go]: Is the `squash` annotation available for `json` ? 🤔

Comment (jeremyrickard) [reply bundle/credentials.go]: Oops sorry, that's not needed :(

File: bundle/credentials.go [hunk 2]
```diff
@@ -0,0 +1,8 @@
+package bundle
+
+// Credential represents the definition of a CNAB credential
+type Credential struct {
+	Location    `mapstructure:",squash"`
```
Comment (carolynvs) [inline bundle/credentials.go]: lol this sent me off on a rabbit hole of "huh, inlining doesn't use 'squash'..." then I found this for the mapstructure lib used by viper https://github.com/mitchellh/mapstructure/issues/42. It got my vote! 👍

File: bundle/credentials_test.go [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestCompleteCredDefinition(t *testing.T) {
+	payload := `{
+		"credentials": {
+			"something": { 
+				"description" : "wicked this way comes",
```
Comment (carolynvs) [inline bundle/credentials_test.go]: 😎

**Pull request merged**